internal/pkg/workspace/workspace_test.go

```diff
@@ -344,6 +344,36 @@ func TestWorkspace_WorkloadExists(t *testing.T) {
 	})
 }
 
+func TestWorkspace_EnvironmentsExist(t *testing.T) {
+	t.Run("returns true if environments are managed in the workspace", func(t *testing.T) {
+		fs := afero.NewMemMapFs()
+		_, _ = fs.Create("/copilot/environments/")
+		ws := &Workspace{
+			copilotDirAbs: "/copilot",
+			fs: &afero.Afero{
+				Fs: fs,
+			},
+		}
+		got, err := ws.HasEnvironment()
+		require.NoError(t, err)
+		require.True(t, got)
+	})
+
+	t.Run("returns false if environments are not managed in the workspace", func(t *testing.T) {
+		fs := afero.NewMemMapFs()
+		_, _ = fs.Create("a/copilot/environments")
+		ws := &Workspace{
```

Comment:
Check out the `TestWorkspace_EnvOverridesPath`, I wonder if we can get to a situation where our unit tests don't rely on setting _any_ private internal fields. I think that makes our unit tests more fragile because whenever we change the internal representation of a type we have to also update our unit tests.

Ideally, we'd be able to write a test like this:
```go
ws, err := Use(fakeFS)
require.NoError(t, err)
ok, err := ws.HasEnvironment()
require.NoError(t, err)
require.True(ok)
```

Reply:
This concerns make sense to me!

My only concern with using `Use`/`Create` in unit test is that it assumes `Use`/`Create` is correctly implemented. The unit test here is then more of an "integration test" that tests whether an exported function works nicely with `Use`  / `Create`. 

I guess this can be thought of a nice perk 🤔

Reply:
done!

Reply:
That's true! It becomes closer to an "integ" test rather than an isolated unit-test 💭 

If it's any consolation the stdlib seems to do the same thing: https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/crypto/sha256/sha256_test.go;l=107

But I agree that it'd be extra nice if only  `Test_Use` were to fail instead of all these other ones in case a change to `Use` breaks a particular expected behavior.